### PR TITLE
Add cmake /build dir to gitignore

### PR DESCRIPTION
**Pull request by pps83**
+ ignore build.* or build-* top level dirs

### CODE REVIEW

**Comment by jkeiser** [comment]
Seems good to me. Maybe a note in the HACKING.md that says "if you name your cmake directories build, build.* or build-*, then they'll be properly ignored, so that's your best bet."

And ... I think we can remove all the executables from .gitignore now!

File: .gitignore [hunk 1]
```diff
@@ -155,6 +155,8 @@ objs
 /tools/json2json
 /tools/jsonstats
 /tools/minify
+/build/
```
Comment (lemire) [inline .gitignore:158]: Can this just be one regex like /build.*/ or something?

Comment (pps83) [reply .gitignore:158]: This aren't regex. Gitignore doest't have full regex support. These two are carefully tested to ensure that they don't ignore some files/folders that happen to have "build" in them. Often, when I build with cmake, I rename build to something like build-rem and regenerate project from scratch

Comment (jkeiser) [reply .gitignore:158]: I believe as long as you prepend the / you only get top level (documentation and my experience so far agree on this point, anyway). `/build*/` would do the trick in that case.

Comment (pps83) [reply .gitignore:158]: Yes, /build is the top level. In my case I actually had to suppress build folders within src tree and **/build*/ would ignore something that I didn't want. For top-level /build*/ should be ok. I can update the PR if you prefer

Comment (jkeiser) [reply .gitignore:158]: Oh, I understand that one. If updating the PR isn't too much out of your way, it seems like the most expedient way forward :)

Comment (pps83) [reply .gitignore:158]: done

Comment (jkeiser) [reply .gitignore]: Thanks for your indulgence :)

**Pull request merged**